src/fetch/FetchExample.re

```diff
@@ -0,0 +1,93 @@
+type tDog = string;
+
+type tArrayDogs = array(tDog);
+
+type state = { dogs: tArrayDogs, isLoading: bool, error: bool };
+
+type action =
+  | DogFetch
+  | DogFetched(tArrayDogs)
+  | DogFailedToFetch;
+
+let component = ReasonReact.reducerComponent("FetchExample");
+
+module Decode = {
+  let dogs = json : tArrayDogs =>
```

Comment:
Is the `Array.map` here really needed? I don't think it should be.

I also think it's clearer written like this:

```ml
let dogs = json : tArrayDogs =>
  Json.Decode.(
    json |> field("message", array(string))
         |> Array.map(dog => dog)
  );
```

Your milage may wary of course, but opening `Json.Decode` locally removes a bit of verbosity, and using the pipe operator `|>` makes the flow of data through the functions more apparent  I think.